src/masternodes/mn_checks.cpp

```diff
@@ -2926,8 +2925,26 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         if (!HasAuth(vault->ownerAddress))
             return Res::Err("tx must have at least one input from token owner");
 
-        if (mnview.GetLoanTokens(obj.vaultId))
-            return Res::Err("Vault <%s> has loans", obj.vaultId.GetHex());
+        if (const auto loans = mnview.GetLoanTokens(obj.vaultId)) {
+            for (const auto& [tokenId, amount] : loans->balances) {
+                const auto rate = mnview.GetInterestRate(obj.vaultId, tokenId, height);
+                if (!rate) {
+                    return Res::Err("Cannot get interest rate for this token (%d)", tokenId.v);
+                }
+                const auto scheme = mnview.GetLoanScheme(vault->schemeId);
+                assert(scheme);
+                const auto tokenInterest = mnview.GetLoanTokenByID(tokenId);
+                assert(tokenInterest);
+                const auto totalInterest = TotalInterest(*rate, height, amount, tokenInterest->interest, scheme->rate);
+                if (totalInterest < 0 && amount + totalInterest <= 0) {
+                    res = mnview.SubMintedTokens(tokenId, amount);
```

Comment:
This is not correct. The token was not taken out of circulation, so it should not be reduced from the minted amount.
negative interest rates reduce the loans, but not the minted token amount.

Reply:
Thanks for spotting this, you are correct.